src/java_bytecode/java_types.h

```diff
@@ -374,4 +374,59 @@ void get_dependencies_from_generic_parameters(
   const typet &,
   std::set<irep_idt> &);
 
+class java_specialized_generic_class_typet : public java_class_typet
+{
+public:
+  /// Build the specialised version of the specific class, with the specified
+  /// parameters and name.
+  /// \param new_tag: The new name for the class (like Generic<java::Float>)
+  /// \param new_components: The specialised components
+  /// \return The newly constructed class.
+  java_specialized_generic_class_typet(
+    const irep_idt &new_tag,
+    const struct_typet::componentst &new_components)
+  {
+    set(ID_C_specialized_generic_java_class, true);
+    // We are specialising the logic - so we don't want to be marked as generic
+    set(ID_C_java_generics_class_type, false);
+    set(ID_name, "java::" + id2string(new_tag));
+    set(ID_base_name, id2string(new_tag));
```

Comment:
Suggest not storing this redundant info unless you really need to

Reply:
My understanding of the specification (and the implementation of generics thus far) is that all of the `base_name`, `name` and `tag` are needed.